pow/src/main/java/tech/pegasys/artemis/pow/event/AbstractEvent.java

```diff
@@ -0,0 +1,30 @@
+/*
+ * Copyright 2019 ConsenSys AG.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except in compliance with
+ * the License. You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under the License is distributed on
+ * an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the License for the
+ * specific language governing permissions and limitations under the License.
+ */
+
+package tech.pegasys.artemis.pow.event;
+
+import tech.pegasys.artemis.pow.api.PowEvent;
+
+public abstract class AbstractEvent<T> implements PowEvent<T> {
+
```

Comment:
In order to use the ServiceAdapter on the other services, does AbstractEvent need to be put in a different package.  Maybe `tech.pegasys.artemis.util.events`? Something like that?